var/spack/repos/builtin/packages/perl-test-warn/package.py

```diff
@@ -10,8 +10,22 @@ class PerlTestWarn(PerlPackage):
     """Perl extension to test methods for warnings"""
 
     homepage = "https://metacpan.org/pod/Test::Warn"
-    url = "http://search.cpan.org/CPAN/authors/id/C/CH/CHORNY/Test-Warn-0.30.tar.gz"
+    url = "https://cpan.metacpan.org/authors/id/B/BI/BIGJ/Test-Warn-0.37.tar.gz"
 
-    license("GPL-1.0-or-later OR Artistic-1.0-Perl")
+    maintainers("EbiArnie")
 
+    license("Artistic-1.0-Perl OR GPL-1.0-or-later")
+
+    version("0.37", sha256="98ca32e7f2f5ea89b8bfb9a0609977f3d153e242e2e51705126cb954f1a06b57")
     version("0.30", sha256="8197555b94189d919349a03f7058f83861f145af9bee59f505bfe47562144e41")
```

Comment:
This version is not available at the new url.  You have a few options depending on what you expect going forward.  The smallest change would be to add the `url` argument to this `version` directive that uses the old URL so it overrides the `url` property.  You could add the `url` argument to all `version` directives (if you think it might change again).  Or you could implement `url_for_version` if you think it is important for unlisted versions (see https://spack.readthedocs.io/en/latest/packaging_guide.html#writing-a-package-recipe for an example).

Reply:
The URL changes when the maintainer changes. In this case this should be rare. Added the URL and also deprecated the older version.